src/lib/util.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const config = require('../config')
+
+/* c8 ignore next 3 */
+function now() {
+  return config.now || new Date().toISOString()
+}
+
+/**
+ * execute tasks in parallel with concurrency
+ * starts on the first "add"
+ * run "concurrency" tasks at time
+ * calls "onTaskComplete" after each task, passing the task's output
+ * stops at first error
+ * does not throw error, return it on "done"
+ */
+function queuedTasks({ concurrency = 1, onTaskComplete = noop } = {}) {
```

Comment:
Typically I reach for `p-queue` for this - it's well used and tested.

Reply:
didn't know that lib, but I'd go with the current one because it's tailored to the needs